crates/types/src/net/mod.rs

```diff
@@ -51,22 +52,57 @@ pub enum AdvertisedAddress {
     /// Unix domain socket
     #[display("unix:{}", _0.display())]
     Uds(PathBuf),
-    /// Hostname or host:port pair, or any unrecognizable string.
+    /// Hostname or host:port pair
```

Comment:
This was actually valid! Thought it could be more specific - particularly, it accepts UDS addresses in the form `unix:<path>`. :-)

Reply:
> form unix:<path>

shouldn't that parse into `Uds(PathBuf)` not `Http(Uri)` ?
and `Http(Uri)` should only accept things like "http://127.0.0.1:5122/"